src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/ProjectDiagnosticSource.cs

```diff
@@ -16,11 +16,10 @@ internal sealed record class ProjectDiagnosticSource(Project Project) : IDiagnos
 {
     public ProjectOrDocumentId GetId() => new(Project.Id);
     public Project GetProject() => Project;
-    public TextDocumentIdentifier GetDocumentIdentifier()
-    {
-        Contract.ThrowIfNull(Project.FilePath);
-        return new VSTextDocumentIdentifier { ProjectContext = ProtocolConversions.ProjectToProjectContext(Project), Uri = ProtocolConversions.GetUriFromFilePath(Project.FilePath) };
-    }
+    public TextDocumentIdentifier? GetDocumentIdentifier()
```

Comment:
Should this be called GetProjectIdentifier? It doesn't do anything with a document here?

Reply:
Or I guess maybe it makes sense because the 'document' that is given is the project itself, OK maybe makes sense then.

Reply:
yes - document in terms of LSP text document